### PR TITLE
CI: update/fix the test workflow

### PR DESCRIPTION
**Pull request by ManuLinares**
- fix the `test_runner` for c3c v0.7.11.
- run the `test.yml` workflow for every push/PR.
- chore: regenerate parser.

### CODE REVIEW

**Comment by ManuLinares** [comment]
@cbuttner When you have the time, could you look at this?

**Review by cbuttner — CHANGES_REQUESTED** [review]
Shouldn't we still test against the test_suite of the latest c3 release instead of the c3c master branch?

**Comment by ManuLinares** [comment]
> Shouldn't we still test against the test_suite of the latest c3 release instead of the c3c master branch?

This is why I asked, having a new pair or eyes. Thanks, I fixed that. Now we are testing always on `latest`.

I removed some files from the ignored list. we still have these:
```
  "test/test_suite/generic/generic_instantiate_alias.c3",
  "test/test_suite/functions/ref_param_fn.c3",
  "test/test_suite/compile_time/ct_switch_more_checks.c3",
  "test/test_suite/struct/const_access_error.c3",
  "test/test_suite/statements/if_single.c3",
  "test/test_suite/statements/do_without_compound.c3",
```

**Comment by ManuLinares** [comment]
@cbuttner  I removed the `test/test_suite/` from the testing. These contain wrong c3 code on purpose.

We test the `latest` release of c3 against stdlib and unit tests instead, which contain over 500 files of valid C3 code.

What do you think?

**Comment by cbuttner** [comment]
test_suite also contains a lot of correct c3 code. We should keep it, it's more code to test against. Also, new C3 features can show up there. It's been very helpful to me in the past.

**Comment by ManuLinares** [comment]
yes but then we must ignore all the cases that contains `// #error`
basically follow what test/src/test_suite_runner.c3 is doing

my point is test_suite isn't about grammar/syntax. is about catching edge cases. We can do add the exceptions to `test_runner/src/main.c3`

Maybe we should. I'll check on this tomorow.

**Comment by cbuttner** [comment]
If there is a tree-sitter parsing error, the compiler is invoked to check if it's meant to be parsed correctly. If the compiler also errors, we can ignore the file. If the compiler parses it fine, it might be an issue with the tree-sitter grammar that has to be investigated. There are some edge cases which are listed in `IGNORED_FILES`. This scheme has been working well for me, I don't see a reason to change it.

**Comment by cbuttner** [comment]
> my point is test_suite isn't about grammar/syntax. is about catching edge cases.

I wouldn't agree, it's testing the parsing, among other things of course. And edge cases is good, we should be testing against those?

**Comment by ManuLinares** [comment]
I agree, it was late and I wasn't thinking clearly.

I'll restore testing on the test_suite. And I'll add these to the ignored: https://github.com/c3lang/tree-sitter-c3/pull/66#issuecomment-4257552034

**Comment by ManuLinares** [comment]
Thanks!
I think fixing the parser for these files is beyond the scope of this PR.

---

```c3
// Files that C3C parses without error but tree-sitter-c3 doesn't
const String[*] IGNORED_FILES = {
	"test/test_suite/generic/generic_instantiate_alias.c3",
	"test/test_suite/functions/ref_param_fn.c3",
	"test/test_suite/compile_time/ct_switch_more_checks.c3",
	"test/test_suite/struct/const_access_error.c3",
	"test/test_suite/statements/if_single.c3",
	"test/test_suite/statements/do_without_compound.c3"
 };
```

**Comment by cbuttner** [comment]
> I think fixing the parser for these files is beyond the scope of this PR.

There is nothing to fix for these files, they are clearly wrong syntax/grammar and won't compile, so the tree-sitter grammar is free to reject them. The point of the ignored files list is that C3C rejects them at a later stage than parsing, so the automatic check doesn't always work.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,51 +1,48 @@
...
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |-
+          gh release download --repo c3lang/c3c --pattern "c3-linux.tar.gz"
+          tar -xzf c3-linux.tar.gz
+          gh release download --repo c3lang/c3c --archive=tar.gz -O c3c-source.tar.gz
+          mkdir -p c3c-source
+          tar -xzf c3c-source.tar.gz -C c3c-source --strip-components=1
+
+      - name: Test stdlib and unit tests
```
Comment (cbuttner) [inline .github/workflows/test.yml]: This name should be changed

File: test_runner/src/main.c3 [hunk 2]
```diff
@@ -4,21 +4,13 @@ import ts, tree_sitter_c3;
...
-  "test/test_suite/define/define_name_errors.c3",
-  "test/test_suite/expressions/addr_of_fails.c3",
-  "test/test_suite/functions/ref_param_fn.c3",
-  "test/test_suite/generic/generic_instantiate_alias.c3",
-  "test/test_suite/generic/generic_mistake_initializer_2099.c3",
-  "test/test_suite/statements/do_without_compound.c3",
-  "test/test_suite/statements/if_single.c3",
-  "test/test_suite/struct/const_access_error.c3",
-};
+	"test/test_suite/generic/generic_instantiate_alias.c3",
```
Comment (cbuttner) [inline test_runner/src/main.c3]: Mixing indentation style here, the files uses spaces.

**Review by cbuttner — CHANGES_REQUESTED** [review]
Please contain the PR to the CI/test_runner changes for clarity (remove generated parser files)

**Comment by ManuLinares** [comment]
> Please contain the PR to the CI/test_runner changes for clarity (remove generated parser files)

Done. Do you regenerate these?

**Comment by cbuttner** [comment]
Thank you.

> > Please contain the PR to the CI/test_runner changes for clarity (remove generated parser files)
> 
> Done. Do you regenerate these?

I can do that and release a new version.
Maybe a Github Action could regenerate the parser automatically, idk.
